equal.go

```diff
@@ -0,0 +1,157 @@
+package ipld
+
+// DeepEqual reports whether x and y are "deeply equal" as IPLD nodes.
+// This is similar to reflect.DeepEqual, but based around the Node interface.
+//
+// Two nodes must have the same kind to be deeply equal.
+// If either node has the invalid kind, the nodes are not deeply equal.
+//
+// Two nodes of scalar kinds (null, bool, int, float, string, bytes)
+// are deeply equal as per the rules of reflect.DeepEqual.
+// In other words, their Go values are compared, as returned by methods such as
+// AsString and AsInt.
+//
+// Links are an exception, since this function does not load links.
+// As such, if either node has the link kind, the nodes are not deeply equal.
```

Comment:
I think this was discussed elsewhere already, but I'd advocate for checking link equality, just in a very literal way -- treating it like a scalar.

I think you can actually use `Link.String()` for this (which will avoid any need to look for pointers vs not, etc).  For CIDs, that should have the effect of ignoring multibase, but regarding the full digest, multicodec indicator, and everything else -- which should be about right.

Reply:
As discussed, I'll use == on these.